clippy_lints/src/useless_conversion.rs

```diff
@@ -67,7 +67,7 @@ impl<'tcx> LateLintPass<'tcx> for UselessConversion {
                 if match_trait_method(cx, e, &paths::INTO) && &*name.ident.as_str() == "into" {
                     let a = cx.typeck_results().expr_ty(e);
                     let b = cx.typeck_results().expr_ty(&args[0]);
-                    if TyS::same_type(a, b) {
+                    if TyS::same_type(a, b) && !adt_different_const_generic_args(&a, &b) {
```

Comment:
`TyS::same_type` is *almost* what we need. So I think we should have our own version of it that includes consts comparisons.

```suggestion
                    if same_type_and_consts(a, b) {
```